cmd/vulcan-nuclei/main.go

```diff
@@ -0,0 +1,269 @@
+		"security-misconfiguration",
+		"generic-detections",
+		"default-credentials",
+		"files",
+		"vulnerabilities",
+		"technologies",
+		"dns",
+		"panels",
+		"tokens",
+	}
+	defaultExclusionList = []string{
+		".github",
+		"fuzzing",
+		"misc",
+		"workflows"}
+)
+
+type void struct{}
+
+type options struct {
+	ForceUpdateTemplates  bool   `json:"force_update_templates"`
+	Severities            string `json:"severities"`
+	TemplateInclusionList string `json:"template_inclusion_list"`
+	TemplateExclusionList string `json:"template_exclusion_list"`
+}
+
+// Vulnerability defines nuclei vulnerability report struct.
+type Vulnerability struct {
+	Name        string `json:"name"`
+	Author      string `json:"author"`
+	Severity    string `json:"severity"`
+	Description string `json:"description"`
+	Matched     string `json:"matched"`
+	Template    string `json:"template"`
+	Type        string `json:"type"`
+	Host        string `json:"host"`
+}
+
+type finding struct {
+	name     string
+	severity string
+	template string
+	match    string
+}
+
+func main() {
+	c := check.NewCheckFromHandler(checkName, run)
+	c.RunAndServe()
+}
+
+func runNucleiCmd(args []string) ([]byte, error) {
+	var err error
+	var cmdOutput []byte
+	err = retry.Do(
+		func() error {
+			cmd := exec.Command(nucleiCmd, args...)
+			cmdOutput, err = cmd.Output()
+			if err != nil {
+				logger.Errorf("exec.Command() with args [%s] failed with %s\nCommand output: %s\n", args, err, string(cmdOutput))
+				return errors.New("nuclei command execution failed")
+			}
+			logger.Infof("nuclei command with args [%s] execution completed successfully", args)
+			return nil
+		},
+		retry.Attempts(3),
+		retry.DelayType(retry.RandomDelay),
+		retry.MaxJitter(5*time.Second),
+	)
+	if err != nil {
+		return []byte{}, err
+	}
+	return cmdOutput, nil
+}
+
+func run(ctx context.Context, target, assetType, optJSON string, state checkstate.State) error {
+	// TODO: If options are "malformed" perhaps we should not return error
+	// but only log and error and return.
+	var opt options
+	if optJSON != "" {
+		if err := json.Unmarshal([]byte(optJSON), &opt); err != nil {
+			return err
+		}
+	}
+
+	isReachable := helpers.IsWebAddrsReachable(target)
+	if !isReachable {
+		return checkstate.ErrAssetUnreachable
+	}
+
+	if opt.ForceUpdateTemplates {
+		_, err := runNucleiCmd([]string{"-update-templates"})
+		if err != nil {
+			return errors.New("nuclei update-templates command execution failed")
+		}
+	}
+
+	availableTemplates := make(map[string]void)
+	logger.Debugf("Templates path: %s", nucleiTemplatePath)
+	tf, err := ioutil.ReadDir(nucleiTemplatePath)
+	if err != nil {
+		return errors.New("unable to read nuclei templates folder")
+	}
+
+	for _, f := range tf {
+		if f.IsDir() {
+			availableTemplates[f.Name()] = void{}
+		}
+	}
+	logger.Infof("Available templates: %+v", availableTemplates)
+
+	selectedTemplates := make(map[string]void)
+	inclusionList := strings.Split(opt.TemplateInclusionList, ",")
```

Comment:
Despite being an extra step, since the options are supplied in JSON, I would prefer to use an array of strings instead of a comma separated string for those, as we do in the ZAP check: https://github.com/adevinta/vulcan-checks/blob/master/cmd/vulcan-zap/manifest.toml#L8

This is not a strong opinion just an observation.

Reply:
Done.